File: CS50_pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>


int main(int argc, string argv[]){
    
    //check if the argc is invalid 
    if(argc !=2){
        printf("error");
        return 1;
    }
    //printing the test to ask for string
    printf("plaintest:");
    string s;
    s=get_string();
    int k;
    k=atoi(argv[1]);
    int n=strlen(s);
     printf("ciphertext:");
    //loop every char in the string and transfrom the letter and kepp other thing the same
    for(int i=0;i<n;i++){
        if(s[i]>='a'&&s[i]<='z'){
            
            
            printf("%c",((int)s[i]+k)%97%26+97);
        }
        else if(s[i]>='A'&&s[i]<='Z'){
            printf("%c",((int)s[i]+k)%65%26+65);
            
        }
        else{
            printf("%c",s[i]);
        }
        
   
        
    }
printf("\n");
return 0;
}

File: CS50_pset6/caesar.py
import cs50
import sys


def main():
    if len(sys.argv) != 2:
        print("error")
        return 1
      
    print("plaintext: ",end='')
    s=cs50.get_string()
    convert(s)

def convert(s):
    # to check if the input the alphabet
    print("ciphertext: ",end='')
    number =int(sys.argv[1])
    for i in s:
        if(i>='a' and i<='z'):
            
            print(chr( (ord(i) + number)%97%26+97),end='')
        elif(i>='A' and i<='Z'):
            
            print(chr( (ord(i) + number)%65%26+65),end='')
        else:
            print(i,end='')
            
            
    
    print('\n', end='')
            
            
            
if __name__ == "__main__":
    main()
 
            
      
            
    
        
        

    